### PR TITLE
Ft create incident 162457681

### PR DESCRIPTION
**Pull request by ogiste**
#### What does this PR do?
Adds endpoint for intervention creation in version 2 of API

#### Description of Task to be completed?
Provide for the creation of valid intervention records within a database

#### How should this be manually tested?
Clone this repo to your machine

 ``` git clone https://github.com/ogiste/ireporter.git ```

Then change the directory to the project by

``` cd ireporter ```

to make sure all the packages needed to run the project present in your machine,
we'll create a virtual enviroment and install the packages there

* to create a virtual enviroment run

    ``` virtualenv -p python3 venv```
* activating the enviroment

    ``` source venv/bin/activate```

The virtual enviroment is now ready, we can install all packages needed for project
ensure you have pip installed otherwise
then on your terminal run

``` pip install -r requirements.txt ```

# run
To test our project on your terminal run

``` export FLASK_APP=run.py```

Export your postgres database url details to the environment:

```
export DB_HOST=localhost
export DB_USER=yourusername
export DB_PASS=yourdbpass
export DB_PORT=5432
export DB_NAME=yourdbname
```

then

``` flask run ```

on your browser open up [http://127.0.0.1:5000/api/v1/incidents](http://127.0.0.1:5000/api/v1/incidents)

# Testing using postman

`POST /api/v2/incidents`

Sample Indicent Record JSON:
  {
            "title":"Corruption",
            "type":"red-flag",
            "location":"-1.324343434, 23.32321323",
            "comment":"Corruption in procurement"
  }

#### What are the relevant pivotal tracker stories?
[#162457681](https://www.pivotaltracker.com/story/show/162457681)

#### Screenshots 
image

### CODE REVIEW

File: app/api/v2/models/incident.py [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+            createdOn=new_incident["createdOn"]
+        )
+        try:
+            self.cursor.execute(insert_incident_statement)
+            incident_results = self.cursor.fetchone()
+            return self.get_single_incident_by_id(incident_results[0])
+        except IntegrityError as e:
+            pprint("Incident model raised an integrity exception: ")
+            if hasattr(e, 'message'):
+                print((e.message))
```
Comment (jmutuku95) [inline app/api/v2/models/incident.py]: remove unnecessary prints

File: app/api/v2/models/incident.py [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+            else:
+                print(e)
```
Comment (jmutuku95) [inline app/api/v2/models/incident.py]: here too

File: app/test/v2/test_incident_views.py [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+        self.msg = {
+            "deleted": "Incident successfully deleted",
+            "created": "Incident successfully created",
+            "updated": "Incident successfully updated",
+            "read": "Incident(s) successfully retrieved",
+            'error': None,
+        }
+
+    def test_incident_create(self):
```
Comment (jmutuku95) [inline app/test/v2/test_incident_views.py]: Add edge cases for create

File: app/api/v2/models/incident.py (hunk 1 above)
Comment (ogiste) [reply app/api/v2/models/incident.py]: Will do.

File: app/test/v2/test_incident_views.py (hunk 3 above)
Comment (ogiste) [reply app/test/v2/test_incident_views.py]: Noted
